include/net/lwm2m.h

```diff
@@ -26,6 +27,43 @@
 #define IPSO_OBJECT_TEMP_SENSOR_ID			3303
 #define IPSO_OBJECT_LIGHT_CONTROL_ID			3311
 
+struct lwm2m_ctx;
+struct lwm2m_message;
+
+/* Establish a message timeout callback type */
+typedef void (*lwm2m_message_timeout_cb_t)(struct lwm2m_message *msg);
```

Comment:
Is lwm2m.h included in doc generation? Because typedef are usually creating doc generation errors if not documented via \@typedef keyword etc...